src/sql/workbench/browser/modal/modal.ts

```diff
@@ -498,9 +498,11 @@ export abstract class Modal extends Disposable implements IThemable {
 				DOM.prepend(this._modalContent, (this._messageElement));
 			}
 		} else {
-			// Set the focus manually otherwise it'll escape the dialog to something behind it
-			this.setInitialFocusedElement();
 			DOM.removeNode(this._messageElement);
+			// Set the focus to first focus element if the focus is not within the dialog
+			if (!DOM.isAncestor(this._bodyContainer.ownerDocument.activeElement, this._bodyContainer)) {
```

Comment:
You should be able to just use document.activeElement. Any reason you're using ownerDocument instead?

Reply:
document.activeElement works too. changed it in latest commit.